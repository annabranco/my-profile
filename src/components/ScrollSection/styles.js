import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorGrayNormal,
  colorWhite,
  colorGrayDark,
  colorBlack,
  fontTitleAlt
} from '../../styles/theme';
import Sea from '../../assets/images/bg/sea.gif';

export const SectionContent = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(
      ${rgba('#ffffff', 0.92)},
      ${rgba('#f0f1fc', 0.93)},
      ${rgba('#e6effb', 0.98)}
    ),
    url(${Sea});
  background-position: center;
  height: auto;
  overflow: hidden;
`;
SectionContent.displayName = 'Section Content';

export const SectionHeader = styled.div`
  z-index: 3;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.3)},
    ${rgba(colorGrayNormal, 0.5)}
  );
  height: 60px;
  width: 100%;
  padding-top: 15px;
  box-shadow: 0 10px 15px 0 ${rgba(colorGrayDark, 0.6)};
  margin: 5px 0 10px;

  &:last-child {
    height: 100px;
  }
`;
SectionHeader.displayName = 'Section Header';

export const Title = styled.h2`
  font-family: ${fontTitleAlt};
  font-size: 2rem;
  color: ${colorBlack};
`;
Title.displayName = 'Title';
