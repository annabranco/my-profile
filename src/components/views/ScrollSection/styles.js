import styled from 'styled-components';
import { rgba } from 'polished';
import {
  colorGrayNormal,
  colorWhite,
  colorGrayDark,
  colorBlack,
  fontTitleAlt
} from '../../../styles/theme';
import Sea from '../../../images/bg/sea.gif';

export const SectionHeaderContainer = styled.div`
  z-index: 3;
  display: flex;
  justify-content: center;
  align-items: flex-start;
  background-image: linear-gradient(
    ${rgba(colorWhite, 0.3)},
    ${rgba(colorGrayNormal, 0.5)}
  );
  height: 60px;
  width: 100%;
  padding-top: 15px;
  box-shadow: 0 10px 15px 0 ${rgba(colorGrayDark, 0.6)};
  margin: 5px 0 20px;

  &:last-child {
    height: 100px;
  }
`;

export const SectionHeaderTitle = styled.h2`
  font-family: ${fontTitleAlt};
  font-size: 2rem;
  color: ${colorBlack};
`;

export const SectionContainer = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-image: linear-gradient(
      ${rgba('#ffffff', 0.92)},
      ${rgba('#f0f1fc', 0.93)},
      ${rgba('#e6effb', 0.98)}
    ),
    url(${Sea});
  background-position: center;
  height: auto;
  overflow: hidden;
  padding-top: 50px;
`;
