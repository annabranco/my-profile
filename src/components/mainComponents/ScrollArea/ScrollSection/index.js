import React from 'react';
import { node, string } from 'prop-types';
import { SectionContent, SectionHeader, Title } from './styles';
import { isDesktop } from '../../../../utils/device';

const ScrollSection = ({ children, title }) => (
  <section id={title}>
    {isDesktop && (
      <SectionHeader>
        <Title>{title}</Title>
      </SectionHeader>
    )}
    <SectionContent>{children}</SectionContent>
  </section>
);

ScrollSection.propTypes = {
  children: node.isRequired,
  title: string.isRequired
};

export default ScrollSection;
