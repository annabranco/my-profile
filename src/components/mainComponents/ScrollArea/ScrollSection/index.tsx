import { ReactElement, ReactNode } from 'react';
import { bool, node, string } from 'prop-types';
import { isDesktop } from '../../../../utils/device';
import { SectionContent, SectionHeader, Title } from './styles';
import { ScrollSectionElement } from '../styles';

interface Props {
  children: ReactNode | ReactNode[];
  datae2eid?: string;
  last: boolean;
  title: string;
}

const ScrollSection = ({
  children,
  datae2eid,
  last,
  title
}: Props): ReactElement => (
  <ScrollSectionElement id={title} last={last}>
    {isDesktop && (
      <SectionHeader>
        <Title data-e2e-id={datae2eid}>{title}</Title>
      </SectionHeader>
    )}
    <SectionContent>{children}</SectionContent>
  </ScrollSectionElement>
);

ScrollSection.propTypes = {
  children: node.isRequired,
  datae2eid: string,
  last: bool,
  title: string.isRequired
};
ScrollSection.defaultProps = {
  datae2eid: null,
  last: false
};
export default ScrollSection;
