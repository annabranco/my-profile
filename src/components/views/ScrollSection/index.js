import React from 'react';
import { node, string } from 'prop-types';
import { SectionContent, SectionHeader, Title } from './styles';

const ScrollSection = ({ children, title }) => (
  <section id={title}>
    <SectionHeader>
      <Title>{title}</Title>
    </SectionHeader>
    <SectionContent>{children}</SectionContent>
  </section>
);

ScrollSection.propTypes = {
  children: node.isRequired,
  title: string.isRequired
};

export default ScrollSection;
