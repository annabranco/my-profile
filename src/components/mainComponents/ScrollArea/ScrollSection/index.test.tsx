import { shallow, ShallowWrapper } from 'enzyme';
import 'jest-styled-components';
import ScrollSection from '.';
import { setupMockProvider } from '../../../../tests/mocks';

const MockChildren = () => <p>Mock Text</p>;

// TODO [10-May -21]: Implement defined testing suite (Anna Branco)

describe('< ScrollSection >', () => {
  let MockProvider;
  let wrapper: ShallowWrapper;

  beforeAll(() => {
    MockProvider = setupMockProvider({ language: 'en' }).MockProvider;

    wrapper = shallow(
      <MockProvider>
        <ScrollSection title="Mock Section">
          <MockChildren />
        </ScrollSection>
      </MockProvider>
    );
  });

  it('should mount', () => {
    expect(wrapper).toBeDefined();
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
