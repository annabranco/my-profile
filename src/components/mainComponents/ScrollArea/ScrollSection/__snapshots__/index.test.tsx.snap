// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< ScrollSection > matches the snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <ScrollSection
    datae2eid={null}
    last={false}
    title="Mock Section"
  >
    <MockChildren />
  </ScrollSection>
</Provider>
`;
